terraspace/app/stacks/peer/main.tf

```diff
@@ -0,0 +1,172 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.38"
+    }
+
+    http = {
+      source  = "hashicorp/http"
+      version = "~> 2.1"
+    }
+  }
+  required_version = ">= 1.0.0"
+}
+
+data "aws_eks_cluster" "eks" {
+  name = module.eks.cluster_id
+}
+
+data "aws_availability_zones" "available" {
+}
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 3.0"
+
+  name                 = var.vpc.name
+  cidr                 = "10.0.0.0/16"
+  azs                  = data.aws_availability_zones.available.names
+  private_subnets      = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24", "10.0.4.0/24"] # Worker Nodes
+  public_subnets       = ["10.0.5.0/24", "10.0.6.0/24", "10.0.7.0/24"]                # LoadBalancer and NAT
+  enable_nat_gateway   = true
+  single_nat_gateway   = true
+  enable_dns_hostnames = true
+
+  public_subnet_tags = {
+    "kubernetes.io/cluster/${var.cluster_name}" = "shared"
+    "kubernetes.io/role/elb"                    = "1"
+  }
+
+  private_subnet_tags = {
+    "kubernetes.io/cluster/${var.cluster_name}" = "shared"
+    "kubernetes.io/role/internal-elb"           = "1"
+  }
+}
+
+module "gateway_endpoint_to_dynamodb" {
+  source         = "../../modules/gateway-endpoint-to-dynamodb"
+  vpc_id         = module.vpc.vpc_id
+  region         = var.region
+  route_table_id = module.vpc.private_route_table_ids[0]
+}
+
+module "gateway_endpoint_to_s3" {
+  source         = "../../modules/gateway-endpoint-to-s3"
+  vpc_id         = module.vpc.vpc_id
+  region         = var.region
+  route_table_id = module.vpc.private_route_table_ids[0]
+}
+
+data "http" "myip" {
+  url = "http://ipv4.icanhazip.com"
+}
```

Comment:
could you add a `todo` here to investigate an alternative approach so we don't have to go through the public internet to achieve this?

Reply:
Done